src/org/usfirst/frc4904/robot/commands/AutoShifter.java

```diff
@@ -0,0 +1,66 @@
+package org.usfirst.frc4904.robot.commands;
+
+
+import java.util.LinkedList;
+import org.usfirst.frc4904.robot.RobotMap;
+import org.usfirst.frc4904.standard.custom.sensors.CustomEncoder;
+import org.usfirst.frc4904.standard.subsystems.chassis.AutoSolenoidShifters;
+import org.usfirst.frc4904.standard.subsystems.chassis.SolenoidShifters;
+import edu.wpi.first.wpilibj.command.Command;
+
+public class AutoShifter extends Command {
+	protected final CustomEncoder leftEncoder;
+	protected final CustomEncoder rightEncoder;
+	protected final AutoSolenoidShifters shifter;
+	protected final LinkedList<Double> speeds = new LinkedList<Double>();
+	public static final double RAPID_ACCELERATION = 8;// WIP
+	public static final double RAPID_DECELERATION = 2;// WIP
+	public static final double RATE_DIFF = 100; // encoder ticks--WIP
+	public static final double SUPER_SLOW_RATE = 2; // ft per sec
+	public static final double MEDIUM_RATE = AutoShifter.SUPER_SLOW_RATE * 2;
+	public static final double FAST_RATE = AutoShifter.SUPER_SLOW_RATE * 4;
+	public static final double SLOW_THROTTLE = 0.4;
+	public static final double MEDIUM_THROTTLE = AutoShifter.SLOW_THROTTLE * 1.875;
+	public static final double FAST_THROTTLE = AutoShifter.SLOW_THROTTLE * 2;
+	public static final double SPEED_LIST_MAX_LENGTH = 5;
+	public static final double LAST_MANUAL_SHIFT_TIME = 5000; // TODO
+	public static final double LAST_AUTO_SHIFT_TIME = 500;// TODO
+
+	public AutoShifter(AutoSolenoidShifters shifter, CustomEncoder leftEncoder, CustomEncoder rightEncoder) {
+		this.shifter = shifter;
+		this.leftEncoder = leftEncoder;
+		this.rightEncoder = rightEncoder;
+	}
+
+	@Override
+	protected void execute() {
+		double speed = (leftEncoder.getRate() + rightEncoder.getRate()) / 2;
+		speeds.add(speed);
+		while (speeds.size() > AutoShifter.SPEED_LIST_MAX_LENGTH) {
+			speeds.removeFirst();
+		}
+		double speedDiffAvg = (speeds.getLast() - speeds.getFirst()) / (speeds.size() - 1);// acceleration
```

Comment:
this does not get you actual acceleration. It is acceleration per tick which is not necessarily accurate. In the future, you should use timestamps.